subsys/bluetooth/controller/hci/hci.c

```diff
@@ -1686,8 +1703,48 @@ static void le_advertising_report(struct pdu_data *pdu_data, u8_t *b,
 		data_len = 0;
 	}
 
+	rssi = b[offsetof(struct radio_pdu_node_rx, pdu_data) +
+		 offsetof(struct pdu_adv, payload) + adv->len];
+
+#if defined(CONFIG_BT_CONTROLLER_PRIVACY)
+	if (direct) {
```

Comment:
This function is getting quite long, so I'd consider refactoring this branch into a helper function. There may be other largish branches that could undergo the same thing.